datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1567,6 +1581,11 @@ fn build_statistics_expr(
                 )),
             ))
         }
+        Operator::LikeMatch => build_like_match(expr_builder).ok_or_else(|| {
+            plan_datafusion_err!(
+                "LIKE expression with wildcard at the beginning is not supported"
+            )
```

Comment:
The idea to return option from `build_like_match` was to avoid communication "i can't derive predicate" via errors.
Changing `build_statistics_expr` to return Option is too much for this PR, but we can maybe return an always-true expression?

Reply:
That's what `unhandled_hook` does. It won't always be a literal `true`. I think it's easiest to follow the pattern in the rest of this file for now and let it bubble up and hit `unhandled_hook`.